Add jc.nvim (#61)
phaazon/this-week-in-neovim-contents: contents/2022/Oct/10.md

---

<h3 id="new-jc.nvim">
    <a href="#new-jc.nvim">
        <span class="icon-text">
            <span class="icon">
                <i class="fa-solid fa-book"></i>
            </span>
            <span>jc.nvim</span>
        </span>
    </a>
</h3>

> jc.nvim – successor of [vim-javacomplete2](https://github.com/artur-shaik/vim-javacomplete2)

Reviewer: ```suggestion
`jc.nvim` – successor of [vim-javacomplete2](https://github.com/artur-shaik/vim-javacomplete2)
```

> which use neovim's built-in LSP client and [Eclipse JDT Language Server](https://github.com/eclipse/eclipse.jdt.ls).

Reviewer: ```suggestion
which uses Neovim's built-in LSP client and [Eclipse JDT Language Server](https://github.com/eclipse/eclipse.jdt.ls).
```

> Main goal of this project is to migrate functionallty of jc2.

Reviewer: ```suggestion
Main goal of this project is to migrate all the functionallties of jc2 to Neovim.
```

> In addition to autocompletion it can:

Reviewer: ```suggestion
In addition to autocompletion, it can:
```

- organize imports;
- generate code (`toString`, `hashCode`, `equals`, constructors, accessors) with field selection;
- add abstract methods to implementing class;
- execute [vimspector](https://github.com/puremourning/vimspector) debug session;
- automatic installation of jdt.ls and java-debug extension;
- class creation methods from `jc2`.

> It's written by [@artur-shaik](https://github.com/artur-shaik) and you read his [blog post](https://shaik.link/posts/javacomplete-to-jc.nvim/) about it.

Reviewer: ```suggestion
It's written by [@artur-shaik](https://github.com/artur-shaik) and you can read his [blog post](https://shaik.link/posts/javacomplete-to-jc.nvim/) about it.
```
